src/business-logic/services/test/device-service.spec.ts

```diff
@@ -1,41 +1,111 @@
-import {DeviceServiceImplementation} from '../device-service-implementation'
-import {Device, TelemetricsDevice} from '../../../model/entities/device'
-import {instance, mock, verify, when} from '@typestrong/ts-mockito'
-import {MongoDeviceRepository} from '../../../data-access/repositories/mongo/mongo-device-repository'
-import {DeviceService} from '../interfaces/device-service'
-import {DeviceEventEmitter} from '../interfaces/device-event-emitter'
-
-describe('DeviceService', () => {
-  let testDevice: TelemetricsDevice
-  let deviceRepository: MongoDeviceRepository
-  let deviceEventEmitter: DeviceEventEmitter
-  const mockedValue: MongoDeviceRepository = mock<MongoDeviceRepository>()
-
-  beforeEach(() => {
-    testDevice = mock<TelemetricsDevice>()
-    deviceRepository = instance(mockedValue)
-    deviceEventEmitter = mock<DeviceEventEmitter>()
+import { DeviceServiceImplementation } from '../device-service-implementation'
+import { Device } from '../../../model/entities/device'
+import { DeviceRepository } from '../../../data-access/repositories/interfaces/device-repository'
+import { instance, mock, verify, when } from '@typestrong/ts-mockito'
+import { DeviceEventEmitter } from '../interfaces/device-event-emitter'
+import { EntityTestFactory } from '../../../model/entities/test/entity-test-factory'
+
+describe(DeviceServiceImplementation.name, () => {
```

Comment:
Since this file is testing `DeviceServiceImplementation` we should update the name of the file to be `device-service-implementation.spec.ts` :)

Reply:
Are there any WebStorm tools that checks consistency between filenames and the classes within?

Reply:
@aardhuis i have not found any linting or other type of rule that can check for this kind of consistency ☹️